source/lesson-2-ruby-and-the-command-line.html.md.erb

```diff
@@ -0,0 +1,722 @@
+  puts("#{x_number} * #{y_number} = #{x_number * y_number}")
+end
+```
+
+</details>
+
+### Conditionals
+
+One last thing to learn is how to get the computer to do something based on
+some condition. To do this, we can use the `if` statement, like so:
+
+```ruby
+puts("What is your name?")
+name = gets()
+if name.include?("r")
+  puts "Hi #{name} - you sound like a pretty cool person"
+else
+  puts "Hi nerd"
+end
+```
+
+An `if` statement takes a "conditional expression" (something that evaluates
+to `true` or `false`). You can use things like `==` (equals), `<` (less than),
+or methods like `.include?()` (like we did in the example).
+
+If the conditional expression is true ruby will run the lines of code between
+the `if` and the `else`, otherwise ruby will run the lines of code between
+the `else` and the `end`.
+
+### Tying it all together
+
+Phew! That was a lot to learn. To recap, we've learned:
+
+* how to use `irb`
+* how to use numbers
+* what "strings" are and how to use them
+* how to use variables to name things
+* how to "interpolate" strings
+* how to call methods
+* how to run ruby from a file
+* how to interact with a user
+* how to define methods
+* how to repeat things
+* how to do things conditionally
+
+If you've understood all of that, well done! You now know the basics of
+programming. There's a lot more to learn, but it all builds on these
+fundamentals.
+
+To tie everything you've learned together, here's one final "hard" task.
+
+#### Task 10: Build a times tables game
+
+Right at the beginning, we said you'd build [this times tables game](https://timestablesgame.richardtowers.repl.run).
+
+To do this you'll need to use everything you've just learned. It's a pretty
+difficult problem for a beginner, so if you get stuck on something don't be
+afraid to ask for help.
+
+Here's a rough plan of how to implement this:
+
+* create a `score` variable to keep track of the users score
+* in a loop that happens 4 times
+  * generate two numbers between 1 and 12 using `rand()` and assign them to variables
+  * print a message to the user asking them to multiply the numbers together
+  * get the user's answer and covert it to a number (using `.to_i`)
+  * compare the user's answer to the answer ruby thinks is correct
+  * show the user a message saying whether they're correct (and add one to their score)
+* if the user scored 4 then show them a message saying they got them all right
+* otherwise, show the user a message commiserating them for their failure
+
+For bonus points, consider whether you could define a method to do some of
+the work to make the loop easier to read.
+
+<details>
+<summary>Answer</summary>
+
+```ruby
+def multiplication_challenge()
+  x = rand(12) + 1
+  y = rand(12) + 1
+  
+  print("What is #{x} multiplied by #{y} ? ")
+  user_input = gets()
+  user_input_as_a_number = user_input.to_i()
+
+  correct_answer = x * y
+  if user_input_as_a_number == correct_answer
+    puts("\u{1F389} correct!")
```

Comment:
This is the first time we've seen `\u{1F389}` (unless I've missed it). I'd be confused and wonder what it's doing, and also want to find out more about how to do it myself. Maybe worth mentioning it somewhere (e.g. `For even more bonus points, try [making your output coloured](link to some resource)`).

Reply:
Ha, `\u{1F389}` is the 🎉 emoji, but you're right, this is totally unnecessary.

`\u{1F643}` is 🙃.

Reply:
Got rid of the unicode, in favour of `:)` and `:(`